### DISCUSSION TITLE
How to use it for blog content

### DISCUSSION THREAD

**Discussion by csantanapr** [squidfunk/mkdocs-material#2573] [Q&A] [answered]
How is the best way to use mkdocs-material as a blog?
Does it have "social" SEO, author info, the typical info that gatsby or jekyll offers for blogging?
Any examples of people using mkdocs as personal website would be welpful.

We are working to migrate (knative.dev](https://knative.dev/) from hugo to mkdocs-material.
We currently host blog post on the project website
Here is the [work in progress](https://julz.github.io/docs/dev/)

**Comment by squidfunk** [comment, marked as answer]
Pretty psyched to see you're considering Material for MkDocs for knative! MkDocs is primarily tailored to technical documentation, i.e. content that is not explicitly tied to dates. By default, it doesn't have author information etc., but this could be added via front matter and by extending the template to render this information, or extracted from git via one of the plugins by @timvink. Besides, there are some plugins that try to add blogging functionality, I haven't tested them though.

Actually, I'm open to adding author information to Material for MkDocs if we could come up with a canonical way of how to do it. Should this be done via front matter? Extracted from git? I'm still unsure what's the best approach that serves the most use cases.

**Reply by timvink** [reply]
@squidfunk Happy to work on a PR to provide native support for [mkdocs-git-authors-plugin](https://github.com/timvink/mkdocs-git-authors-plugin). In an earlier discussion we decided to leave it out of scope (https://github.com/squidfunk/mkdocs-material/issues/1403), but it makes sense when improving support for blogging.

@csantanapr For blogging, you might want to try [mkdocs-rss-plugin](https://pypi.org/project/mkdocs-rss-plugin/). And, in case you haven't found it in your research, checkout this [overview of mkdocs plugins](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins).

**Reply by squidfunk** [reply]
@timvink - I'm still looking for a one-stop solution, if possible. Why split the git information among so many plugins? 😉

**Reply by csantanapr** [reply]
Thanks for the quick response. 

I don't think we need git author information, we put the author info in the md file at the top. 

Here is the latest blog for releases announcement of the software https://github.com/knative/docs/blob/main/blog/releases/announcing-knative-v0-21-release.md

We use Hugo and the directory structure and file name is use to create the menu and sort. 

But I guess the menu can be compose manually. 

Thanks for sharing rss the website today has rss with Hugo we will need it with mkdocs

Rss is important as I know of many companies including mine that we have slack hook to the release announcement blog posts. 

I would think a releases announcements page and rss would be something common for open source website/docs we want mkdocs to be the blog and website in addition of the docs refs

**Reply by timvink** [reply]
> Why split the git information among so many plugins?

Lot of reasons, like latency, do 1 thing cleanly, legacy, dependencies. But mainly: I don't have the time or motivation to spend a lot of time trying to build one git-based plugin to rule them all :) The situation is also not so bad: [mkdocs-git-revision-date-localized-plugin](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin) for created and updated dates, [mkdocs-git-authors-plugin](https://github.com/timvink/mkdocs-git-authors-plugin) for author information, [mkdocs-git-committers-plugin](https://github.com/byrnereese/mkdocs-git-committers-plugin) for git information from the github API, and you've covered almost all use cases.

**Reply by squidfunk** [reply]
Thanks for the explanation!

**Comment by squidfunk** [comment]
As a first step to provide better support for blogging, tags support is going to land soon in #2593!

**Reply by csantanapr** [reply]
Hi @squidfunk it would better for us to support front matter that way we can just the current blog over the new mkdocs

**Comment by csantanapr** [comment]
Just wanted to provide an update on progress we were able to migrate the blog from hugo to mkdocs and our version is located here
https://knative.dev/blog/ and includes rss

There are few things that we would like to see is how to leverage the front matter to display authors avatar, name, and link and also use the date field to sort and shortTitle or tittle to create the navigation menu automatically

**Reply by squidfunk** [reply]
@csantanapr it would be great to have a list of requirements from users. I’ve asked for this quite a few times but haven’t seen any so far. It would help me to make sure that a native implementation can tick all boxes.

**Reply by csantanapr** [reply]
@squidfunk  I can contribute to the list of requirements. Where should I list them?
I can help with the lost and point out the the things that we would like see support lit of the box, like a good snippet plugin that respect the layout of the parent page

**Reply by squidfunk** [reply]
Please create a new issue and group your requirements in must have and nice to have, because coming from Hugo there might be a lot of them. I would then provide basic support first, adding new functionality as we go.

**Reply by squidfunk** [reply]
I've started to collect some ideas in #3353. Please post your input there as a comment.
